src/Hosting/WindowsServices/src/ServiceBaseLifetime.cs

```diff
@@ -1,41 +1,41 @@
-﻿#if NET461
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
 using System;
 using System.ServiceProcess;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
 
-namespace GenericHostSample
+namespace Microsoft.Extensions.Hosting.WindowsServices
 {
-    public static class ServiceBaseLifetimeHostExtensions
-    {
-        public static IHostBuilder UseServiceBaseLifetime(this IHostBuilder hostBuilder)
-        {
-            return hostBuilder.ConfigureServices((hostContext, services) => services.AddSingleton<IHostLifetime, ServiceBaseLifetime>());
-        }
-
-        public static Task RunAsServiceAsync(this IHostBuilder hostBuilder, CancellationToken cancellationToken = default)
-        {
-            return hostBuilder.UseServiceBaseLifetime().Build().RunAsync(cancellationToken);
-        }
-    }
-
     public class ServiceBaseLifetime : ServiceBase, IHostLifetime
     {
         private TaskCompletionSource<object> _delayStart = new TaskCompletionSource<object>();
 
-        public ServiceBaseLifetime(IApplicationLifetime applicationLifetime)
+        public ServiceBaseLifetime(IHostingEnvironment environment, IApplicationLifetime applicationLifetime, ILoggerFactory loggerFactory)
         {
+            Environment = environment ?? throw new ArgumentNullException(nameof(environment));
             ApplicationLifetime = applicationLifetime ?? throw new ArgumentNullException(nameof(applicationLifetime));
+            Logger = loggerFactory.CreateLogger("Microsoft.Hosting.Lifetime");
```

Comment:
Curious why we went the LoggerFactory route instead of something like https://github.com/aspnet/AspNetCore/blob/3fd8a97af2e8cf2885a8d5836cb5d4f199458d22/src/Hosting/Hosting/src/Internal/ApplicationLifetime.cs#L21

Reply:
In this case we wanted to pick a specific provider name that's common across several components.